File: entities/blackhole.py
import pygame
import math
from config import *

class BlackHole:
    def __init__(self, x, y, radius):
        self.x = x
        self.y = y
        self.radius = radius

    def update(self, mouse_pos):
        self.x, self.y = mouse_pos

    def draw(self, screen):
        pygame.draw.circle(screen, BLACK, (int(self.x), int(self.y)), self.radius)

File: src/BlackHole.cpp
#include "BlackHole.h"
#include "config.h"

BlackHole::BlackHole(float x, float y) : pos(x, y) {
    radius = BLACKHOLE_SIZE;
    shape.setRadius(radius);
    shape.setOrigin(radius, radius); // Set origin to the center for easy positioning
    shape.setFillColor(sf::Color::Black);
}

void BlackHole::update(sf::Vector2f mousePos) {
    this->pos = mousePos;
    shape.setPosition(this->pos);
}

void BlackHole::draw(sf::RenderWindow& window) {
    window.draw(shape);
}
